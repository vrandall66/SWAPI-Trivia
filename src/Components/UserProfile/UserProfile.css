.UserProfile {
  background-color: lightgrey;
  display: flex;
  flex-direction: column;
  margin: 0;
  max-width: 600px;
  position: absolute;
  right: 0;
  z-index: 1;
}
.UserProfile__button--sign-out {
  background-color: green;
}

.UserProfile--userName {
  margin: 0;
}

.UserProfile__hamburger-icon {
  height: 42px;
  margin: 0;
  width: 42px;
}

.UserProfile--userMenu {
  display: flex;
  justify-content: space-between;
  width: 100%;
}