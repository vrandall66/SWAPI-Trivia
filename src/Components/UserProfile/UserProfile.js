import React from "react";
import { Link } from "react-router-dom";
import "./UserProfile.css";
import hamburgerIcon from "../../images/HamburgerIcon.png";

const UserProfile = ({
  isOpen,
  userName,
  userFavQuote,
  userRanking,
  userFavCharacters,
  updateUserMenuState,
  userLogoutReset
}) => {
  return (
    <div className="UserProfile">
      <div className="UserProfile--userMenu">
        <h1 className="UserProfile--userName">{userName}</h1>
        <input
          type="image"
          alt="User Profile Dropdown Menu Icon"
          src={hamburgerIcon}
          className="UserProfile__hamburger-icon"
          onClick={updateUserMenuState}
        />
      </div>
      {isOpen ? (
        <div>
          <h5>{userFavQuote}</h5>
          <h5>{userRanking}</h5>
          <h5>{userFavCharacters}</h5>
          <Link to="/">
            <button
              type="button"
              className="UserProfile__button--sign-out"
              onClick={userLogoutReset}
            >
              Sign out
            </button>
          </Link>
        </div>
      ) : null}
    </div>
  );
};

export default UserProfile;
