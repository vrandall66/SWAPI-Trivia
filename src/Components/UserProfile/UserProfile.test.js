import React from "react";
import { shallow } from "enzyme";
import UserProfile from "./UserProfile";

describe("UserProfile", () => {
  let wrapper;
  let isOpen = false;
  let name = "Matt";
  let userFavQuote = "I know";
  let userRanking = "Intermediate";
  let userFavCharacters = [];
  let updateUserMenuState = jest.fn();
  let userLogoutReset = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <UserProfile
        isOpen={isOpen}
        userName={name}
        userFavQuote={userFavQuote}
        userRanking={userRanking}
        userFavCharacters={userFavCharacters}
        updateUserMenuState={updateUserMenuState}
        userLogoutReset={userLogoutReset}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
