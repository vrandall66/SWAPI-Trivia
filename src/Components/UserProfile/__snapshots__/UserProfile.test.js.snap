// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot 1`] = `
<div
  className="UserProfile"
>
  <div
    className="UserProfile--userMenu"
  >
    <h1
      className="UserProfile--userName"
    >
      Matt
    </h1>
    <input
      alt="User Profile Dropdown Menu Icon"
      className="UserProfile__hamburger-icon"
      onClick={[MockFunction]}
      src="inactiveHamMenu.png"
      type="image"
    />
  </div>
</div>
`;
